src/useE2eeConfig.ts

```diff
@@ -0,0 +1,56 @@
+/*
+Copyright 2023 New Vector Ltd
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import { useMemo } from "react";
+import { logger } from "matrix-js-sdk/src/logger";
+
+import { E2EEConfig } from "./livekit/useLiveKit";
+
+export const useE2eeConfig = (
+  passwordString: string | null
+): E2EEConfig | undefined => {
+  return useMemo(() => {
+    if (!passwordString) return undefined;
+
+    // Since crypto.getRandomValues() gives UInt8Arrays, we get a 128 bit buffer and end up
+    // base64 encoding it to get a text string. It's a bit silly to then pass this string to
```

Comment:
A couple words explaining what makes it silly would be welcome (since that was the part I was skeptical about)